File: Nov2020/28Nov/N-CoOrdinates(Map Practice).cpp
/*
Problem Name: N - Co Ordinates (Map Practice)
Problem Link: https://www.hackerearth.com/practice/data-structures/hash-tables/basics-of-hash-tables/practice-problems/algorithm/n-co-ordinates-map-practice/
*/

#include<bits/stdc++.h>
using namespace std;
int main(){
    ios::sync_with_stdio(0);    
    cin.tie(0);             
    int n,x,y;
    cin>>n;
    map<pair<int,int>,int> m;
    for(int i=0;i<n;++i){
        cin>>x>>y;
        m[{x,y}]++;
    }
    for(auto l:m){
        cout<<l.first.first<<" "<<l.first.second<<" "<<l.second<<"\n";
    }
    return 0;
}

File: Nov2020/28Nov/N-CoOrdinates(MapPractice).java
/*
Problem Name: N - Co Ordinates (Map Practice)
Problem Link: https://www.hackerearth.com/practice/data-structures/hash-tables/basics-of-hash-tables/practice-problems/algorithm/n-co-ordinates-map-practice/
*/
import java.util.*;
import java.io.*;
class TestClass {
    public static void main(String args[] ) throws Exception {
        Scanner in = new Scanner(System.in);
        PrintWriter out = new PrintWriter(System.out);
        int n = in.nextInt();
        TreeMap<Integer,TreeMap<Integer,Integer>> map = new TreeMap<Integer,TreeMap<Integer,Integer>>();
        for(;n>0;n--){
            int x = in.nextInt(), y = in.nextInt();
            if(map.containsKey(x)){
                if(map.get(x).containsKey(y))
                    map.get(x).put(y, map.get(x).get(y) + 1);
                else
                    map.get(x).put(y, 1);
            }
            else{
                TreeMap<Integer,Integer> innerMap = new TreeMap<Integer,Integer>();
                innerMap.put(y, 1);
                map.put(x,innerMap);
            }
        }
        for(int x: map.keySet()){
            for(int y: map.get(x).keySet())
                out.println(x+" "+y+" "+map.get(x).get(y));
        }
        out.flush();
    }
}
